### ISSUE TITLE
Error in my Owncloud and files SMB pending

### ISSUE THREAD

**Issue by diogogomes101** [owncloud/core#25916]
### Steps to reproduce
1.
2.
3.

### Expected behaviour
The size SMB folders ok and no errors in my logs

### Actual behaviour
The size is pending and i have errors in my logs.

strpos() expects parameter 1 to be string, array given at /var/www/owncloud/settings/admin.php#201

### Server configuration
**Operating system**:
Ubuntu 16.04.1

**Web server:**
apache2
**Database:**
mysql
**PHP version:**
7.0
**ownCloud version:** (see ownCloud admin page)
9.2
**Updated from an older ownCloud or fresh install:**
fresh install

Login as admin user into your ownCloud and access 
http://example.com/index.php/settings/integrity/failed 

No errors have been found.

**List of activated apps:**

If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your ownCloud installation folder

Enabled:
  - activity: 2.3.2
  - comments: 0.3.0
  - dav: 0.2.5
  - federatedfilesharing: 0.3.0
  - federation: 0.1.0
  - files: 1.5.1
  - files_external: 0.6.0
  - files_pdfviewer: 0.8.1
  - files_sharing: 0.10.0
  - files_texteditor: 2.1
  - files_trashbin: 0.9.0
  - files_versions: 1.3.0
  - files_videoplayer: 0.9.8
  - firstrunwizard: 1.1
  - notifications: 0.3.0
  - provisioning_api: 0.5.0
  - systemtags: 0.3.0
  - updatenotification: 0.2.1
Disabled:
  - encryption
  - external
  - files_antivirus
  - gallery
  - templateeditor
  - user_external
  - user_ldap

**The content of config/config.php:**

<?php
$CONFIG = array (
  'updatechecker' => false,
  'instanceid' => 'oc8e1edc796d',
  'passwordsalt' => '',
  'secret' => '',
  'trusted_domains' =>
  array (
    0 => '192.168.1.4',
  ),
  'datadirectory' => '/var/www/owncloud/data',
  'overwrite.cli.url' => 'http://192.168.1.4/owncloud',
  'dbtype' => 'mysql',
  'version' => '9.2.0.1',
  'dbname' => 'owncloud-db',
  'dbhost' => 'localhost',
  'dbtableprefix' => 'oc_',
  'dbuser' => 'oc_admin',
  'dbpassword' => '',
  'logtimezone' => 'UTC',
  'installed' => true,
  'loglevel' => 0,
);

**Are you using external storage, if yes which one:** local/smb/sftp/...
YES, smb on windows server 2012 r2

**Are you using encryption:** yes/no
no

**Are you using an external user-backend, if yes which one:** LDAP/ActiveDirectory/Webdav/...
no

#### Web server error log
```
Insert your webserver log here
```

#### ownCloud log (data/owncloud.log)
```
Insert your ownCloud log here

Error	PHP	strpos() expects parameter 1 to be string, array given at /var/www/owncloud/settings/admin.php#201	2016-08-23T12:11:43+00:00
Error	PHP	strpos() expects parameter 1 to be string, array given at /var/www/owncloud/settings/admin.php#216	2016-08-23T12:11:43+00:00
Error	PHP	strpos() expects parameter 1 to be string, array given at /var/www/owncloud/settings/admin.php#211	2016-08-23T12:11:43+00:00
Error	PHP	strpos() expects parameter 1 to be string, array given at /var/www/owncloud/settings/admin.php#206	2016-08-23T12:11:43+00:00
Error	PHP	strpos() expects parameter 1 to be string, array given at /var/www/owncloud/settings/admin.php#201	2016-08-23T12:11:43+00:00
Error	PHP	strpos() expects parameter 1 to be string, array given at /var/www/owncloud/settings/admin.php#216	2016-08-23T12:11:43+00:00
Error	PHP	strpos() expects parameter 1 to be string, array given at /var/www/owncloud/settings/admin.php#211	2016-08-23T12:11:43+00:00
Error	PHP	strpos() expects parameter 1 to be string, array given at /var/www/owncloud/settings/admin.php#206	2016-08-23T12:11:43+00:00
Error	PHP	strpos() expects parameter 1 to be string, array given at /var/www/owncloud/settings/admin.php#201	2016-08-23T12:11:43+00:00
Error	PHP	strpos() expects parameter 1 to be string, array given at /var/www/owncloud/settings/admin.php#216	2016-08-23T12:11:43+00:00
```

My problem is the folders are configure SMB the size is pending.
I have the 2 virtual machines to test this problem.

In my server witch owncloud 9.1.0 the subfolders dissapear when cron.php run.
In my server witch owncloud 9.2.0 this problem not occour because my cron is ajax.

After install owncloud i run this commands to smb authentication works:

apt-get install smbclient
apt-get install libsmbclient-dev
git clone git://github.com/eduardok/libsmbclient-php.git
cd libsmbclient-php ; phpize
./configure
make
sudo make install
make install INSTALL_ROOT=/tmp/smbc
vi /etc/php5/apache2/php.ini Add to the file -> extension="smbclient.so"
service apache2 restart

Am I doing something wrong?

Thanks

**Comment by PVince81** [comment]
Please make sure your cron is configured and running properly. There is a background job that will scan the SMB storage through cron. You can also force an initial scan with `occ files:scan --all`.

The error in the log are unrelated to SMB, I'll make a separate ticket for it

**Comment by PVince81** [comment]
New ticket for the warnings: https://github.com/owncloud/core/issues/25925

**Comment by diogogomes101** [comment]
The occ files:scan --all not working to solve my problem. =(
Why the auto run cron.php in my owncloud 9.1 the subfolders smb disapear in owncloud?
Please help-me

**Comment by PVince81** [comment]
Please note that this is a bug tracker, not a support platform. See http://owncloud.org/support

Try `occ files:scan --all -vvvvv` and post details.
Are you using the "save in session" node ?

**Comment by diogogomes101** [comment]
This command not scan my smb folder.

        Folder /diogo/thumbnails/18058
        File   /diogo/thumbnails/18058/32-32.png
        File   /diogo/thumbnails/18058/2048-2048-max.png
        Folder /diogo/thumbnails/18541
        File   /diogo/thumbnails/18541/32-32.png
        File   /diogo/thumbnails/18541/2048-2048-max.png

And in the log :

Error	PHP	strpos() expects parameter 1 to be string, array given at /var/www/owncloud/settings/admin.php#216	2016-08-24T09:21:49+00:00
Error	PHP	strpos() expects parameter 1 to be string, array given at /var/www/owncloud/settings/admin.php#211	2016-08-24T09:21:49+00:00
Error	PHP	strpos() expects parameter 1 to be string, array given at /var/www/owncloud/settings/admin.php#206	2016-08-24T09:21:49+00:00
Error	PHP	strpos() expects parameter 1 to be string, array given at /var/www/owncloud/settings/admin.php#201	2016-08-24T09:33:32+00:00
Error	PHP	strpos() expects parameter 1 to be string, array given at /var/www/owncloud/settings/admin.php#216	2016-08-24T09:33:32+00:00
Error	PHP	strpos() expects parameter 1 to be string, array given at /var/www/owncloud/settings/admin.php#211	2016-08-24T09:33:32+00:00
Error	PHP	strpos() expects parameter 1 to be string, array given at /var/www/owncloud/settings/admin.php#206	2016-08-24T09:33:32+00:00
Error	PHP	strpos() expects parameter 1 to be string, array given at /var/www/owncloud/settings/admin.php#201	2016-08-24T09:33:32+00:00
Error	PHP	strpos() expects parameter 1 to be string, array given at /var/www/owncloud/settings/admin.php#216	2016-08-24T09:33:32+00:00
Error	PHP	strpos() expects parameter 1 to be string, array given at /var/www/owncloud/settings/admin.php#211	2016-08-24T09:33:32+00:00
Error	PHP	strpos() expects parameter 1 to be string, array given at /var/www/owncloud/settings/admin.php#206	2016-08-24T09:33:32+00:00
Error	PHP	strpos() expects parameter 1 to be string, array given at /var/www/owncloud/settings/admin.php#201	2016-08-24T09:33:32+00:00

**Comment by PVince81** [comment]
Are you using the "save in session" node ?

**Comment by diogogomes101** [comment]
what is the "save in session" ?

**Comment by PVince81** [comment]
I mean "Log-in credentials, save in session".
Maybe post a screenshot of your external storage settings (and censor private details if needed)

**Comment by diogogomes101** [comment]
No, save in session.

![image](https://cloud.githubusercontent.com/assets/21194487/17936383/8abc0c58-6a16-11e6-8aa9-2aca5770c0a4.png)

**Comment by diogogomes101** [comment]
to smb configure after install owncloud i can run this commands.

apt-get install smbclient
apt-get install libsmbclient-dev
git clone git://github.com/eduardok/libsmbclient-php.git
cd libsmbclient-php ; phpize
./configure
make
sudo make install
make install INSTALL_ROOT=/tmp/smbc
vi /etc/php5/apache2/php.ini Add to the file -> extension="smbclient.so"
service apache2 restart


If I install only apt-get install smbclient the smb not working. 

is this normal?

**Comment by PVince81** [comment]
> No, save in session.

Do you mean "No, I am indeed using 'save in session'  mode" or "No save in session used in my setup" ? (without comma).

With "smbclient" alone, the connection to some Windows Servers do not seem to work correctly. It is best to always have the PHP module "php5-libsmbclient" installed for best results.

Is your smbclient able to connect to your SMB server from the CLI of the ownCloud server ?

**Comment by diogogomes101** [comment]
No save in session used in my setup.
the smbclient not connect in my windows server.
to connect i can run this commands.

apt-get install smbclient
apt-get install libsmbclient-dev
git clone git://github.com/eduardok/libsmbclient-php.git
cd libsmbclient-php ; phpize
./configure
make
sudo make install
make install INSTALL_ROOT=/tmp/smbc
vi /etc/php5/apache2/php.ini Add to the file -> extension="smbclient.so"
service apache2 restart

I run php7.0

**Comment by PVince81** [comment]
> to connect i can run this commands.

Do you mean that the command line smbclient can now correctly connect to your SMB server ?
If yes, then ownCloud should also be able to.

**Comment by diogogomes101** [comment]
yes, now smbclient connect smb server, but after running the commands.

**Comment by PVince81** [comment]
then problem solved ?

**Comment by diogogomes101** [comment]
no.
Files SMB pending and when run cron.php every 15 minutes the subfolders in my share desappear in onwcloud.

**Comment by ownclouders** [comment]
Hey, this issue has been closed because the label `needs info` is set and there were no updates for 14 days. Feel free to reopen this issue if you deem it appropriate.

(This is an automated comment from [GitMate.io](https://gitmate.io).
